ceed-fortran.c

```diff
@@ -320,27 +319,26 @@ static int CeedQFunction_n = 0;
 static int CeedQFunction_count_max = 0;
 
 struct fContext {
-  void (*f)(void *ctx, void *qdata, int *nq,
+  void (*f)(void *ctx, int *nq,
             const CeedScalar *const u,const CeedScalar *const u1,const CeedScalar *const u2,
             CeedScalar *const v1,CeedScalar *const v2, int *err);
   void *innerctx;
 };
 
-static int CeedQFunctionFortranStub(void *ctx, void *qdata, int nq,
+static int CeedQFunctionFortranStub(void *ctx, int nq,
                                     const CeedScalar *const *u, CeedScalar *const *v) {
   struct fContext *fctx = ctx;
   int ierr;
 
   CeedScalar ctx_=1.0;
-  fctx->f((void*)&ctx_, qdata, &nq, u[0], u[1], u[4], v[0], v[1], &ierr);
+  fctx->f((void*)&ctx_, &nq, u[0], u[1], u[2], v[0], v[1], &ierr);
```

Comment:
I am concerned about this approach handling up to 16 input fields and 16 output fields. I'm not sure what would be a comparable way in Fortran to handle an array of CeedScalar arrays like we do with indata and outdata in the ref implementation in C, but 32 inputs feels like a bad way to do it.

Reply:
It is possible (but not portably in F77) to pass arrays of pointers. But most Fortran programmers are used to very long argument lists. I know there are CESM components where >30 argument functions are common.

Cc: @thilinarmtb

Reply:
Thanks for the ping @jedbrown ! For the time being this is okay.  We can try
to figure out a way around this if possible.